src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -1,4 +1,37 @@
 package com.kodilla.ecommercee.controller;
 
+import com.kodilla.ecommercee.entity.GenericEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping(value = "carts", produces = "application/json")
 public class CartController {
-}
+
+    @PostMapping()
+    public void createCart() {
+
+    }
+
+    @GetMapping("{cartId}")
+    public List<GenericEntity> getProducts(@PathVariable Long cartId) {
+        return new ArrayList<>();
+    }
+
+    @PostMapping("{cartId}/products")
+    public GenericEntity addProducts(@PathVariable Long cartId, @RequestBody List<GenericEntity> productList) {
+        return new GenericEntity("addProducts test");
+    }
+
+    @DeleteMapping("{cartId}/products/{productId}")
+    public boolean removeProduct(@PathVariable Long cartId, @PathVariable Long productId) {
+        return true;
+    }
+
+    @PostMapping("{cartId}/createOrder")
+    public void createOrder(@PathVariable Long cartId) {
+
+    }
```

Comment:
Czy ta metoda nie pokrywa się z tą w OrderController? Tak się zastanawiam. Wydaje mi się, że Order powstaje na podstawie cartId ale w kontrolerze Order...

Reply:
Ja też bym ta metodę zostawiła w Order

Reply:
@ewaradomska Proponuję
![image](https://user-images.githubusercontent.com/32686501/64173379-6f856300-ce57-11e9-8ad5-125723455b55.png)
a logikę za utworzeniem Orderu zostawić w Cart'cie (tak jak jest na liście endpointów)

Reply:
dobra, biorę to :P pytanie ewentualnie czy jest sens powielać tą metodę... czy ewentualnie na poziomie orderControllera zrobić po prostu metodę, która umożliwia wpisanie z palca body do OrderDto (np. kiedy zamówienie jest tworzone na poziomie "sklepu" bo załóżmy klient telefonicznie zamawia jedną rzecz, albo zamówienie jest niestandardowe) i wtedy wrzucić @RequestBody OrderDto jako parametr... (ale to będziemy dyskutować jak wrzucę rzeczywistą implementację OrderControllera, bo sobie ją tam dziergam).